Mathlib/Topology/ContinuousMap/Interval.lean

```diff
@@ -0,0 +1,176 @@
+/-
+Copyright (c) 2024 Vincent Beffara. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Vincent Beffara
+-/
+import Mathlib.Topology.CompactOpen
+import Mathlib.Topology.Order.ProjIcc
+
+/-!
+# Continuous bundled maps on intervals
+
+In this file we prove a few results about `ContinuousMap` when the domain is an interval.
+-/
+
+open Set ContinuousMap Filter Topology
+
+namespace ContinuousMap
+
+variable {α : Type*} [LinearOrder α] [TopologicalSpace α] [OrderTopology α] {a b c : α}
+variable {E F : Type*} [TopologicalSpace E] [TopologicalSpace F]
+
+/-- The embedding into an interval from a sub-interval lying on the left, as a `ContinuousMap`. -/
+def subinterval_left (h : b ∈ Icc a c) : C(Icc a b, Icc a c) where
+  toFun x := ⟨x, x.2.1, x.2.2.trans h.2⟩
+
+/-- The embedding into an interval from a sub-interval lying on the right, as a `ContinuousMap`. -/
+def subinterval_right (h : b ∈ Icc a c) : C(Icc b c, Icc a c) where
+  toFun x := ⟨x, h.1.trans x.2.1, x.2.2⟩
+
+/-- The value of a `ContinuousMap` defined on an interval at the left extremity of the interval. -/
+def leftval (hab : a ≤ b) (f : C(Icc a b, E)) : E :=
+  f ⟨a, le_rfl, hab⟩
```

Comment:
Is this not
```suggestion
def leftval (hab : a ≤ b) (f : C(Icc a b, E)) : E := f ⊥
```
once you have made `Set.Icc.orderBot` a local instance? Do you think we need it as a separate definition?

Reply:
OK it is simpler like you say, definition not needed.